docs/user/adding-new-metrics.md

```diff
@@ -1,97 +1,216 @@
+  
+- `application`: The metric is related to an application run, and is cleared every time the application restarts.
   This should be used for things that are constant during the run of an application, such as the operating system version.
   In practice, these metrics are generally set during application startup.
   A common mistake---using the `ping` lifetime for these type of metrics---means that they will only be included in the first ping sent during a particular run of the application.
+  
 - `user`: The metric is part of the user's profile.
   This should be used for things that change only when the user's profile is created.
   It is rare to use this lifetime outside of some metrics that are built in to Glean, such as `client_id`.
 
-### Naming things
+While lifetimes are important to understand for all metric types, they are particularly important for the metric types that record single values and don't aggregate on the client (`boolean`, `string`, `labeled_string`, `string_list`, `datetime` and `uuid`), since these metrics will send the "last known" value and missing the earlier values could be a form of unintended data loss.
+
+#### A lifetime example
+
+Let's work through an example to see how these lifetimes play out in practice. Let's suppose we have a user preference, "turbo mode", which defaults to `false`, but the user can turn it to `true` at any time.  We want to know when this flag is `true` so we can measure its affect on other metrics in the same ping.  In the following diagram, we look at a time period that sends 4 pings across two separate runs of the application. We assume here, that like Glean's built-in [metrics ping](pings/metrics.html), the developer writing the metric isn't in control of when the ping is submitted. 
+
+In this diagram, the ping "lifetimes" are represented as rectangles, but the moment the ping is "submitted" is represented by its right edge. The user changes the "turbo mode" setting from `false` to `true` at 3 seconds.
+  
+```mermaid
+gantt
+  axisFormat %S
+  title Metric lifetimes
+  dateFormat  s
+
+  section Metrics<br/>pings
+    Ping 1 :active, p1, 0, 2s
+    Ping 2 :active, p2, 2, 3s
+    Ping 3 :active, p3, 5, 10s
+    Ping 4 :active, p4, 15, 5s
+    Ping 5 :active, p5, 20, 5s
+
+  section Process
+	Run 1     :done, a1, 0, 10s
+	Run 2     :done, a2, 15, 10s
 
-Metric identifiers (the combination of a metric's category and name) must be unique across all metrics that are sent by an application.
+  section Interaction
+    ↓ Turns on turbo mode : 3
+
+  section Ping<br/>(change)
+	True      :3, 2s
+
+  section Ping<br/>(init and change)
+    False     :0, 2s
+    True      :3, 2s
+    True      :15, 5s
+
+  section Application<br/>(change)
+    True      :3, 12s  
+
+  section Application<br/>(init and change)
+    False     :0, 3s
+    True      :3, 12s
+    True      :15, 10s
+
+  section User<br/>(change)
+    True      :3, 22s
+```
```

Comment:
![image](https://user-images.githubusercontent.com/38294/75825259-ff5a8a00-5d72-11ea-9e03-e2a65af50980.png)

Reply:
Can we break out this lifetime example part to a separate PR?

Reply:
It's a little mixed up, particularly since the lifetime section was moved as part of this change.  Maybe we can do another round of review all together, and if it gets stalled, I'll pick the two apart.